src/pages/HomePage.js

```diff
@@ -1,80 +1,122 @@
-import React, {useState} from 'react';
+import React, { useState } from 'react';
 import { UrlForm } from '../components/UrlForm';
 import ReactMarkdown from 'react-markdown';
 
-
-
 const HomePage = () => {
-
-  const FETCH_URL = 'http://localhost:8000'
-
-  // set state for urls
+  const FETCH_URL = 'http://localhost:8000';
+  
   const [urls, setUrls] = useState({
     url1: '',
     url2: ''
   });
-
-  // set state for comparison result
+  
+  const [preferences, setPreferences] = useState({
+    selected_categories: [],
+    user_preference: ''
+  });
+  
   const [comparison, setComparison] = useState('');
-
-  // change states for urls
+  const [error, setError] = useState('');
+  
   const handleChange = (e) => {
-    const {name, value} = e.target;
-    setUrls((prevUrls) => ({
-      ...prevUrls,
-      [name]: value,
-    }));
-  }
+    const { name } = e.target;
+    
+    if (name === 'url1' || name === 'url2') {
+      setUrls(prevUrls => ({
+        ...prevUrls,
+        [name]: e.target.value,
+      }));
+    } else if (name === 'selected_categories') {
+      const selectedOptions = Array.from(e.target.selectedOptions);
+      const selectedValues = selectedOptions.map(option => option.value);
+      setPreferences(prevPreferences => ({
+        ...prevPreferences,
```

Comment:
If you want the preferences state to reset and have no default values every time selected_categories is updated, then you would not include ...prevPreferences.

Reply:
I was treating these two new fields the same as the urls, which aren't reset after the submit `Compare` is pressed. My thinking was that a user may want to modify the comparison they just made. We could assume they don't want that, and reset the fields. Or we could provide a `Reset` button to clear the form.